### PR TITLE
refactor: migrate verilog terminal from window global + DOM to Pinia synthesisStore

### LINKED ISSUE #1054: refactor: Migrate Verilog terminal output from window global + DOM to Pinia store
## Summary

The Verilog terminal currently communicates between `Verilog2CV.js` (simulator layer) and `VerilogEditorPanel.vue` (Vue layer) through two anti-patterns:

1. **`window.verilogTerminal`** : a global reference to the Vue component instance, set in `VerilogEditorPanel.vue` 
2. **`document.getElementById('verilogOutput')`** : raw DOM manipulation as a fallback when the Vue component hasn't mounted yet

There is an existing TODO in the codebase acknowledging this:

```js
// TODO: It needs to be handled using pinia after moving it to vue components(Verilog2CV.js L93)
```

## Problem

1. **Race condition:** If synthesis fires before `VerilogEditorPanel.vue` mounts, `window.verilogTerminal` is `undefined` and messages are silently dropped
2. **Tight coupling:** The simulator layer directly manipulates DOM elements (`innerHTML`, `.style.color`), bypassing Vue's reactivity
3. **Not testable:** `window` globals and `document.getElementById` cannot be mocked cleanly in Vitest

## Proposed Solution

Create a new `synthesisStore.ts` to handle terminal messages and synthesis state, keeping `verilogStore.ts` untouched (it stays responsible for terminal visibility and theme only).

## Context

This is a prerequisite for the GSoC Project 7 (Client-Side Verilog Synthesis) implementation, which will extend this pattern for synthesis status, progress, and compilation metrics.

### PR DESCRIPTION
**Pull request by Me-Priyank**
Fixes #1054

 

#### Describe the changes you have made in this PR -
Replaces `window.verilogTerminal` global and `document.getElementById('verilogOutput')` DOM manipulation with a new `synthesisStore.ts` Pinia store. Terminal messages now flow reactively through the store instead of through window globals and raw DOM writes. Resolves the existing TODO in `Verilog2CV.js`
- **New:** `src/store/synthesisStore.ts` - `messages`, `addMessage()`, `clearMessages()`
- **Modified:** `Verilog2CV.js`, `VerilogTerminal.vue`, `VerilogEditorPanel.vue`
- No visual changes - pure internal refactor

### Screenshots of the UI changes (If any) -
N/A — no UI changes.
---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Terminal output now auto-scrolls when new compilation messages appear.

* **Refactor**
  * Terminal now uses a centralized message store for consistent display and clearing across the app.
  * Programmatic parent-level terminal control APIs were removed; terminal behavior is managed via the shared store.

* **Chores**
  * Verilog editor theme changes now apply automatically.

[Review Change Stack](https://app.coderabbit.ai/change-stack/CircuitVerse/cv-frontend-vue/pull/1055?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src-tauri/Cargo.toml [hunk 1]
```diff
@@ -15,13 +15,14 @@ name = "app_lib"
 crate-type = ["staticlib", "cdylib", "rlib"]
 
 [build-dependencies]
-tauri-build = { version = "2.0.5", features = [] }
+tauri-build = { version = "2", features = [] }
```
[the remark this answers is not in the document]
Comment (Me-Priyank) [reply src-tauri/Cargo.toml]: Cargo.lock is committed in this PR. you can check the latest commit 964b023

**Comment by Me-Priyank** [comment]
@Nihal4777 @ThatDeparted2061 @092vk ready for review

**Comment by 092vk** [comment]
@Me-Priyank, please squash all the commits into one commit.

**Comment by Me-Priyank** [comment]
> @Me-Priyank, please squash all the commits into one commit.

Done @092vk

**Comment by Me-Priyank** [comment]
@Nihal4777 @ThatDeparted2061 @092vk PTAL

File: src/store/synthesisStore.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { defineStore } from "pinia";
+import { ref } from "vue";
+
+export interface SynthesisMessage {
+  text: string;
+  type: "info" | "error" | "success";
```
Comment (Nihal4777) [inline src/store/synthesisStore.ts]: It's good to declare a reusable message type to avoid duplication.

File: src/simulator/src/Verilog2CV.js
Comment (Nihal4777) [inline src/simulator/src/Verilog2CV.js:1]: semicolons can be added

File: src/simulator/src/Verilog2CV.js [hunk 3]
```diff
@@ -70,35 +71,13 @@ export function applyVerilogTheme(theme) {
...
 
 function setVerilogOutput(text, type = 'info') {
-    if (typeof window !== 'undefined' && window.verilogTerminal) {
-        window.verilogTerminal.addMessage(text, type)
-    } else {
-        const verilogOutputDiv = document.getElementById('verilogOutput')
-        if (verilogOutputDiv) {
-            if (type === 'error') {
-                verilogOutputDiv.innerHTML = text
-                verilogOutputDiv.style.color = '#ff6b6b'
```
Comment (Nihal4777) [inline src/simulator/src/Verilog2CV.js:80]: is this color scheme preserved? @Me-Priyank @coderabbitai

File: src/store/synthesisStore.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const addMessage = (text: string, type: SynthesisMessageType = "info") => {
+    messages.value.push({ text, type, timestamp: new Date() });
+  };
+
+  const clearMessages = () => {
+    messages.value = [];
+  };
+
+  return {
+    messages,
```
Comment (Nihal4777) [inline src/store/synthesisStore.ts]: if messages are not updated then they can be made readonly.

File: src/simulator/src/events.ts
Comment (Nihal4777) [inline src/simulator/src/events.ts:1]: Are few files pushed by mistake?

Comment (Me-Priyank) [reply src/simulator/src/events.ts:1]: Yes, those files were accidental churn from the earlier branch/conflict-resolution state. I first removed them from the PR, but the CI format check then failed because `oxfmt --check "**/*.ts"` runs repo-wide and reported those files.

I have now kept them only as format-only changes to satisfy CI. The actual Verilog terminal Pinia migration remains limited to the Verilog panel, terminal, Verilog2CV, and synthesis store files.

File: src/store/synthesisStore.ts (hunk 4 above)
Comment (Me-Priyank) [reply src/store/synthesisStore.ts]: Yes, that makes sense.

This is closely related to my Week 4/5 deliverables around the Pinia synthesis store and reactive Verilog terminal PTAL [HERE](https://docs.google.com/document/d/1XJqjMMRzUpk63Moq_f3iqC5A1Ex06TqVVjUihb4x_V4/edit?pli=1&tab=t.0) , because at that stage the terminal state should have a clear store contract: components should read terminal messages, while updates should happen only through store actions like `addMessage()` and `clearMessages()`.

Since this PR already introduces the initial `synthesisStore`, I’ll handle the minimal version here itself by exposing `messages` as readonly. That keeps the current change safer without waiting for the later store cleanup work. :))

File: src/store/synthesisStore.ts (hunk 2 above)
Comment (Me-Priyank) [reply src/store/synthesisStore.ts]: Done @Nihal4777 . Thanks for suggestion :))

File: src/store/synthesisStore.ts (hunk 4 above)
Comment (Me-Priyank) [reply src/store/synthesisStore.ts]: Done.

File: src/simulator/src/Verilog2CV.js
Comment (Me-Priyank) [reply src/simulator/src/Verilog2CV.js:1]: Done

Comment (Nihal4777) [reply src/simulator/src/Verilog2CV.js:1]: @Me-Priyank, Missing in imports.. same for other files.

Comment (Me-Priyank) [reply src/simulator/src/Verilog2CV.js:1]: Done @Nihal4777 PTAL

File: src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue [hunk 5]
```diff
@@ -30,23 +30,30 @@
...
-    type: 'info' | 'error' | 'success'
-    timestamp: Date
-}
+import { ref, onUnmounted, nextTick, watch } from 'vue';
+import { useVerilogStore } from '../../../store/verilogStore';
+import { useSynthesisStore } from '../../../store/synthesisStore';
 
 const verilogStore = useVerilogStore()
+const synthesisStore = useSynthesisStore()
 const terminalHeight = ref(200)
```
Comment (Nihal4777) [inline src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue]: @Me-Priyank missing semi-colons

Comment (Me-Priyank) [reply src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue]: Done... added the missing semicolons consistently, not only in imports.

**Review by Nihal4777 — APPROVED** [review]
LGTM!

File: src/components/Panels/VerilogEditorPanel/VerilogEditorPanel.vue
Comment (ThatDeparted2061) [inline src/components/Panels/VerilogEditorPanel/VerilogEditorPanel.vue]: Seems dead after all, writes to this were removed. So shudnt this be deleted

File: src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue [hunk 6]
```diff
@@ -30,128 +30,108 @@
...
+import { useVerilogStore } from '../../../store/verilogStore';
+import { useSynthesisStore } from '../../../store/synthesisStore';
 
-interface Message {
-    text: string
-    type: 'info' | 'error' | 'success'
-    timestamp: Date
-}
+const verilogStore = useVerilogStore();
+const synthesisStore = useSynthesisStore();
```
Comment (ThatDeparted2061) [inline src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue:38]: useSynthesisStore() is called inside every setVerilogOutput() / clearVerilogOutput() call, if either fires before Pinia boots, it throws "getActivePinia was called with no active Pinia". Hoist the store instance to module scope.

**Review by ThatDeparted2061 — CHANGES_REQUESTED** [review]
document.getElementById called twice per usage in verilogModeSet (5 times in the function), (window as any).editor global still present, no unit tests for the new store, and ~85% of the diff is unrelated formatting churn that will cause conflicts with #1060.

**Comment by ThatDeparted2061** [comment]
> document.getElementById called twice per usage in verilogModeSet (5 times in the function), (window as any).editor global still present, no unit tests for the new store, and ~85% of the diff is unrelated formatting churn that will cause conflicts with #1060.

just flagging it up

File: src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue (hunk 6 above)
Comment (ThatDeparted2061) [reply src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue:38]: Pinia composables must be called after app.use(pinia) has run. If either setVerilogOutput or clearVerilogOutput is invoked during module initialization, this throws "getActivePinia was called with no active Pinia". Please hoist the store instance to module scope
```js
const store = useSynthesisStore()

export function setVerilogOutput(...) {
  store.setOutput(...)
}
```

File: src/components/Panels/VerilogEditorPanel/VerilogEditorPanel.vue
Comment (Me-Priyank) [reply src/components/Panels/VerilogEditorPanel/VerilogEditorPanel.vue]: Yes correct ... removed

File: src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue (hunk 6 above)
Comment (Me-Priyank) [reply src/components/Panels/VerilogEditorPanel/VerilogTerminal.vue:38]: Fixed... hoisted to a lazy-initialized module-scoped cache via `getSynthesisStore()`

**Comment by Me-Priyank** [comment]
> document.getElementById called twice per usage in verilogModeSet (5 times in the function), (window as any).editor global still present, no unit tests for the new store, and ~85% of the diff is unrelated formatting churn that will cause conflicts with #1060.

- Those double `getElementById` / `(window as any).editor` both patterns exist identically on upstream/main and are not introduced by this PR. The diff on those lines is semicolons only from the formatting pass, but i cleaned it up .... `verilogModeSet` now reuses the already-queried const instead of calling `getElementById`/`querySelector` twice per element.

- Added 9 tests for `synthesisStore` covering `addMessage`, `clearMessages`, type defaults, timestamps, and accumulation. Thanks for flagging this

- About those formatting churn, as noted in my earlier comment, removing those files fails CI because `oxfmt --check` runs repo-wide.  #1060 carries the same formatting changes for the same reason, so there's no actual conflict between them on those files.

**Comment by ThatDeparted2061** [comment]
Cool lemme look at it by the end of the day

**Review by ThatDeparted2061 — APPROVED** [review]
